### PR TITLE
New color schemes

### PR DESCRIPTION
**Pull request by couet**
# This Pull request:
Choosing an appropriate color scheme is essential for making results easy to understand and
interpret. Factors like colorblindness and converting colors to grayscale for publications
can impact accessibility. Furthermore, results should be aesthetically pleasing. The following
three color schemes, recommended by M. Petroff in [arXiv:2107.02270v2](https://arxiv.org/pdf/2107.02270)
and available on [GitHub](https://github.com/mpetroff/accessible-color-cycles)
under the MIT License, meet these criteria.

## Changes or fixes:
- Implement the new three colors scheme in TColor.cxx et RTypes.h
- Add 3 missing colors (kBrown, kGrape and kAsh)
- Automaticallically set the name of a new color to its hexadecimal value
- Implement two new tutorials illustrating these new color schemes
- Update the doc
- Improve the dark and bright colors management. Modify the documentation accordingly.

## Checklist:

- [X] tested changes locally
- [X] updated the docs (if necessary)

### CODE REVIEW

**Comment by couet** [comment]
The test failure on mac13 is not due to the changes of this PR.

**Comment by lenzip** [comment]
Looks good to me, thanks!

**Comment by DuesselbergAdrian** [comment]
> The test failure on mac13 is not due to the changes of this PR.

So some old svg tests (o) are failing because of a changed fill attribute:
- Original: fill="#917f6e"
- Modified: fill="#6f2da8"

The json tests (j) are failing because of differences in fOper.

Do we need new reference files ?

File: core/base/src/TColor.cxx [hunk 1]
```diff
@@ -194,6 +195,25 @@ boxes (see TWbox, TPave, TPaveText, TPaveLabel, etc).
...
+## Accessible Color Schemes
+Choosing an appropriate color scheme is essential for making results easy to understand and
+interpret. Factors like colorblindness and converting colors to grayscale for publications
+can impact accessibility. Furthermore, results should be aesthetically pleasing. The following
+three color schemes, recommended by M. Petroff in [arXiv:2107.02270v2](https://arxiv.org/pdf/2107.02270)
+and available on [GitHub](https://github.com/mpetroff/accessible-color-cycles)
+under the MIT License, meet these criteria.
+
+These three color schemes are available as color sets with 6, 8, and 10 colors, named
+`kP[6, 8, 10]ColorName`. For example, `kP6Red` represents the red color within the P6 color scheme.
```
[the remark this answers is not in the document]
Comment (couet) [reply core/base/src/TColor.cxx:208]: P = Petroff the author of these Schemes

**Comment by couet** [comment]
> > The test failure on mac13 is not due to the changes of this PR.
> 
> So some old svg tests (o) are failing because of a changed fill attribute:
> 
> * Original: fill="#917f6e"
> * Modified: fill="#6f2da8"
> 
> The json tests (j) are failing because of differences in fOper.
> 
> Do we need new reference files ?

May be @linev can help ?

**Comment by linev** [comment]
@couet  

This PR introduces new colors. 
TWebCanvas always transfer list of existing colors to the client - and coded it in compact form into `fOper` field of some JSON object. Thats why it changes.

> So some old svg tests (o) are failing because of a changed fill attribute:

This need to be investigated. Most probably some of new colors overrides already existing one and therefore fill attributes of some drawing changes.

**Comment by couet** [comment]
There was some overwriting while running stressGraphics, but this has been fixed. Now, the new colors do not overwrite any existing ones. I carefully checked using TColor::ListColors to ensure I chose unused indices.
Could it be that Web Graphics uses colors not seen by ListColors ?

**Comment by linev** [comment]
As I can see from log output - difference are in **OLD** svg files, generated not with the JSROOT but with TSVG. 
Like here:

```
166, 166: <path d="M216.21,421.836l 12, -12h 125v -160l 12, -12v 184h -149z" fill="#917f6e"/>  <->  <path d="M216.21,421.836l 12, -12h 125v -160l 12, -12v 184h -149z" fill="#6f2da8"/>
  207, 207: <path d="M429.154,421.836l 12, -12h 100v -160l 12, -12v 184h -124z" fill="#917f6e"/>  <->  <path d="M429.154,421.836l 12, -12h 100v -160l 12, -12v 184h -124z" fill="#6f2da8"/>
  248, 248: <path d="M357.251,382.288l 3, -3h 13v -99l 3, -3v 105h -19z" fill="#67494e"/>  <->  <path d="M357.251,382.288l 3, -3h 13v -99l 3, -3v 105h -19z" fill="#92dadd"/>
  255, 255: <path d="M418.092,382.288l 3, -3h 13v -99l 3, -3v 105h -19z" fill="#67494e"/>  <->  <path d="M418.092,382.288l 3, -3h 13v -99l 3, -3v 105h -19z" fill="#92dadd"/>
  SVG test failed for archi
```

**Comment by couet** [comment]
Does that mean the ref files are produced with TSVG ? It sounds a bit strange to me ... TSVG is old code and should not be used as a reference.

**Comment by couet** [comment]
TSVG (old code) is like TPDF or TPostScript is simply producing an SVG file from what is displayed it does not create a new color.

**Comment by linev** [comment]
> it does not create a new color.

But somehow new color affects SVG production.

**Comment by couet** [comment]
only on Mac ? ok I will check

**Comment by DuesselbergAdrian** [comment]
> only on Mac ? ok I will check TSVG

The tests are only enabled on Mac

**Comment by couet** [comment]
`arrows.C` is one of the faulty examples in this PR. I have generated the SVG files (using TSVG) with and without the changes proposed in this PR. The files are the same! no color overwriting whatsoever......

**Comment by DuesselbergAdrian** [comment]
> `arrows.C` is one of the faulty examples in this PR. I have generated the SVG files (using TSVG) with and without the changes proposed in this PR. The files are the same! no color overwriting whatsoever......

Can you please try this with hstack.C, archi.C, basic3d.C or greyscale.C. Arrows.C is only failing for jsons

**Comment by couet** [comment]
ok I will try hstack.C

**Comment by couet** [comment]
ok I see a difference in the lego plot.
Screenshot 2024-09-04 at 17 35 02

File: core/base/src/TColor.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcanal) [reply core/base/src/TColor.cxx:208]: This should be clarified in the text. (albeit it could also be `P` for `Prefered` :) ).

**Comment by dpiparo** [comment]
Same comments of this PR https://github.com/root-project/root/pull/16372 apply here. The graphics tests have to be updated.

**Comment by couet** [comment]
I see there is still a couple of `_o` test failing. I am investigating.

**Comment by couet** [comment]
It is green... is Sb can approve

**Comment by couet** [comment]
It is all green. Can somebody approve it?

**Review by dpiparo — APPROVED** [review]
lgtm

**Pull request merged**